go/private/actions/archive.bzl

```diff
@@ -125,12 +128,19 @@ def emit_archive(ctx, go_toolchain, mode=None, golib=None, goembed=None, direct=
         objects = extra_objects,
         archive = archive,
     )
-  return GoArchive(
-      mode = mode,
+  data = GoArchiveData(
       file = out_lib,
+      importpath = importpath,
       searchpath = searchpath,
-      library = golib,
+  )
+  return GoArchive(
+      mode = mode,
+      data = data,
       embed = goembed,
       direct = direct,
```

Comment:
`direct` is still a list of `GoArchive`s, right? Is `GoArchive` ever added to a depset? If so, we might still get bad hash performance for archives in the DAG that are reachable on multiple paths. This might affect other providers, too.

Reply:
GoArchive is never allowed in a depset, only GoArchiveData